src/main/java/eu/dissco/core/translator/configuration/RabbitMQConfiguration.java

```diff
@@ -0,0 +1,20 @@
+package eu.dissco.core.translator.configuration;
+
+import eu.dissco.core.translator.component.MessageCompressionComponent;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class RabbitMQConfiguration {
+
+  @Bean
+  public RabbitTemplate compressedTemplate(ConnectionFactory connectionFactory,
```

Comment:
Where is connectionFactory defined? Don't you need a bean?

Reply:
Spring inject this, similar to the constructor, if a bean is available.
We might be able to use this elsewhere as well
https://docs.spring.io/spring-framework/reference/core/beans/java/bean-annotation.html#beans-java-dependencies